ghga_connector/core/session.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Handling seesion initialization for requests"""
+
+import requests
+from requests.adapters import HTTPAdapter, Retry
+
+
+class RequestsSession:
+    """Helper clas to make max_retries user configurable"""
+
+    session: requests.Session
+
+    @classmethod
+    def get(cls, *args, **kwargs):
+        """Delegate to session method"""
+        return cls.session.get(*args, **kwargs)
+
+    @classmethod
+    def patch(cls, *args, **kwargs):
+        """Delegate to session method"""
+        return cls.session.patch(*args, **kwargs)
+
+    @classmethod
+    def post(cls, *args, **kwargs):
+        """Delegate to session method"""
+        return cls.session.post(*args, **kwargs)
+
+    @classmethod
+    def put(cls, *args, **kwargs):
+        """Delegate to session method"""
+        return cls.session.put(*args, **kwargs)
+
+
+def configure_session(max_retries: int):
```

Comment:
This could even be a class method of the RequestSession and could then be called `configure`. Just because configure is tightly coupled to that class, however, the current setup is also fine.

Reply:
Moved the actual logic to a `configure` classmethod, though kept the function as thin wrapper, so configuration sites don't need to know about the class